src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/DuplicateWaysCheck.java

```diff
@@ -0,0 +1,115 @@
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * This check looks for partially or completely duplicated Ways via Edges.
+ *
+ * @author savannahostrowski
+ */
+public class DuplicateWaysCheck extends BaseCheck
+{
+    // You can use serialver to regenerate the serial UID.
+    private static final long serialVersionUID = 1L;
+    public static final String DUPLICATE_EDGE_INSTRUCTIONS = "This way, {0,number,#}, "
+            + "has at least one duplicate segment.";
+
+    public static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList(DUPLICATE_EDGE_INSTRUCTIONS);
+    public static final int ZERO_LENGTH = 0;
+    public static final String AREA_KEY = AreaTag.KEY;
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public DuplicateWaysCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge
+                // Check to see that the Edge is a master Edge
+                && ((Edge) object).isMasterEdge()
+                // Check to see that the edge has not already been seen
+                && !this.isFlagged(((Edge) object).getMasterEdgeIdentifier())
+                // Check to see that the edge is car navigable
+                && HighwayTag.isCarNavigableHighway(object)
+                // The edge is not part of an area
+                && !object.getTags().containsKey(AREA_KEY);
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Set<Segment> allEdgeSegments = new HashSet<>();
+        final Edge edge = (Edge) object;
+
+        final Rectangle bounds = edge.asPolyLine().bounds();
+        // Get Edges which are contained by or intersect the bounds, and then filter
+        // Out the non-master Edges as the bounds Edges are not guaranteed to be uni-directional
+        final Iterable<Edge> edgesInBounds = Iterables
+                .stream(edge.getAtlas().edgesIntersecting(bounds)).filter(Edge::isMasterEdge);
+
+        for (final Edge edgeInBounds : edgesInBounds)
+        {
+            // If the Edge found in the bounds has an area tag or if the Edge has a length of 0
+            // Or if the Edge has already been flagged before then continue because we don't want to
+            // Flag area Edges or duplicate Nodes
+            if (edgeInBounds.getTags().containsKey(AREA_KEY)
+                    || edgeInBounds.asPolyLine().length().equals(Distance.meters(ZERO_LENGTH))
+                    || this.isFlagged(edgeInBounds.getMasterEdgeIdentifier()))
+            {
+                continue;
+            }
+
+            // If the Set of Edges does not contain the Edge found in the bounds
+            if (!allEdgeSegments.containsAll(edgeInBounds.asPolyLine().segments()))
```

Comment:
Please consider using [`PolyLine.intersects` method](https://github.com/osmlab/atlas/blob/c0965af0fd1563844fff505a2a80cc79b95f655e/src/main/java/org/openstreetmap/atlas/geography/PolyLine.java#L592) to check for an overlap. You can also utilize [`PolyLine.intersections` method](https://github.com/osmlab/atlas/blob/c0965af0fd1563844fff505a2a80cc79b95f655e/src/main/java/org/openstreetmap/atlas/geography/PolyLine.java#L570) to get overlapping locations.